### PR TITLE
Interactive delete

### LINKED ISSUE #854: Interactive deletion
``atuin search --delete -i`` should work as an interactive deletion.
Right now the ``--delete`` argument is ingored.

### PR DESCRIPTION
**Pull request by YummyOreo**
closes #854 

This implements if you press `Alt + d` on a entry. It will delete all of the entries matching that exactly.
> Note: this keybinding is subject to comments and change. 

To do this I added a `Strict` search mode. 

Todo:
- [ ] Add some sort of confirmation
- [ ] Only enable when deletion is enabled
- [ ] Improve ui (should not exit when after deletion)

### CODE REVIEW

File: atuin/src/command/client/search/interactive.rs [hunk 1]
```diff
@@ -574,15 +593,43 @@ pub async fn history(
...
+            while !entries.is_empty() {
+                for entry in &entries {
+                    println!("deleting {}", entry.command);
+                    db.delete(entry.clone()).await?;
+                }
+
+                entries =
+                    run_non_interactive(&settings, OptFilters::default(), query, &mut db).await?;
+            }
+            Ok(String::new())
```
Comment (YummyOreo) [inline atuin/src/command/client/search/interactive.rs:632]: Pull this out into a function that both deletes can use

File: atuin-client/src/settings.rs [hunk 2]
```diff
@@ -50,6 +54,7 @@ impl SearchMode {
             SearchMode::FullText if settings.search_mode == SearchMode::Skim => SearchMode::Skim,
             // otherwise fuzzy.
             SearchMode::FullText => SearchMode::Fuzzy,
+            SearchMode::Strict => SearchMode::Strict,
```
Comment (YummyOreo) [inline atuin-client/src/settings.rs:57]: Should not be the same

Comment (YummyOreo) [reply atuin-client/src/settings.rs:57]: Mby fuzzy/skim

**Comment by ellie** [comment]
This is now part of the inspector! Closing 🙏

**Pull request closed without merging**